### PR TITLE
CIRC-616: Add raml and request schema for the declare item lost API

### PR DESCRIPTION
**Pull request by mrdpopov**

### CODE REVIEW

File: ramls/declare-item-lost.raml [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+
+traits:
+  validate: !include raml-util/traits/validation.raml
+
+/circulation:
+  /loans/{id}:
+    /declare-item-lost:
+      post:
+        is: [validate]
+        body:
```
Comment (bohdan-suprun) [inline ramls/declare-item-lost.raml]: Could you please link the RAML with schema file.

It seems you need:
```
body:
    application/json:
        type: !declare-item-lost-request.json
```

File: ramls/declare-item-lost-request.json [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+  "type": "object",
+  "description": "Request body schema for declare item lost API",
+  "properties": {
+    "declaredLostDateTime": {
+      "description": "The date and time when the item to be declared lost",
+      "type": "string",
+      "format": "date-time"
+    },
+    "comment": {
+      "description": "Comment to the declare item lost action stored in the loan history",
```
Comment (marcjohnson-kint) [inline ramls/declare-item-lost-request.json]: That this comment is stored in the loan action history is specific to the implementation. Please could this description describe the intent of the comments. Maybe it is `stored` part of this statement that is the source of my discomfort.

```suggestion
      "description": "Comment explaining why the item has been declared lost",
```

File: ramls/declare-item-lost-request.json [hunk 3]
```diff
@@ -0,0 +1,19 @@
+{
+  "$schema": "http://json-schema.org/draft-04/schema#",
+  "type": "object",
+  "description": "Request body schema for declare item lost API",
+  "properties": {
+    "declaredLostDateTime": {
+      "description": "The date and time when the item to be declared lost",
```
Comment (marcjohnson-kint) [inline ramls/declare-item-lost-request.json]:
```suggestion
      "description": "The date and time when the item is to be declared lost",
```

File: ramls/declare-item-lost-request.json (hunk 2 above)
Comment (mrdpopov) [reply ramls/declare-item-lost-request.json]: This was taken from the similar functionality where comments were needed. I've changed it to lower the level of discomfort of anyone who reads it.

File: ramls/declare-item-lost-request.json (hunk 3 above)
Comment (mrdpopov) [reply ramls/declare-item-lost-request.json]: Sure. Typo.

File: ramls/declare-item-lost.raml [hunk 4]
```diff
@@ -0,0 +1,36 @@
+#%RAML 1.0
+title: Declare item lost
+version: v0.1
```
Comment (marcjohnson-kint) [inline ramls/declare-item-lost.raml]: Is this the same as the version in the module descriptor?

**Pull request merged**